File: Scripts/volatile_and_force_compiler_task/json/perl/json.pl
use JSON;

my $r = 1;

sub executeTask {
	my ($i) = @_;

	my $data = decode_json('{ "foo": 1, "bar": [10, "apples"] }');
	my $sample = { blue => [1,2], ocean => "water" };
	my $json_string = encode_json($sample);
	return ($i + 1);
}

for ( $a = 0; $a < 10000000; $a = $a + 1) {
	$r = executeTask($a + $r);
}

File: scripts/volatile_and_force_compiler_task/json/php/json.php
<?php

$r = 1;

function executeTask($i) {

	$data = json_decode('{ "foo": 1, "bar": [10, "apples"] }'); // dictionaries will be returned as objects
	$sample = array( "blue" => array(1,2), "ocean" => "water" );
	$json_string = json_encode($sample);
	return $i + 1;
}

for ( $i = 0; $i < 10000000; $i++ ) {
	$r = executeTask($i + $r);
}
?>
